File: app/src/main/java/me/kirimin/mitsumine/db/FeedDAO.java
package me.kirimin.mitsumine.db;

import java.util.Calendar;
import java.util.List;
import java.util.Locale;

import com.activeandroid.query.Delete;
import com.activeandroid.query.Select;
import me.kirimin.mitsumine.model.Feed;

public class FeedDAO {

    public static void save(Feed feed) {
        Calendar cal = Calendar.getInstance(Locale.JAPAN);
        feed.saveTime = cal.getTime().getTime();
        feed.save();
    }

    public static List<Feed> findAll() {
        return new Select().from(Feed.class).execute();
    }

    public static List<Feed> findByType(String type) {
        return new Select().from(Feed.class).where("type = ?", type).execute();
    }
    
    public static void deleteOldData(){
        Calendar cal = Calendar.getInstance(Locale.JAPAN);
        long fiveDays = 1000 * 60 * 60 * 24 * 3;
        new Delete().from(Feed.class).where("saveTime < ? AND type = ?", cal.getTime().getTime() - fiveDays, Feed.TYPE_READ).execute();
    }

    private FeedDAO() {
    }
}

File: app/src/main/java/me/kirimin/mitsumine/db/FeedDAO.kt
package me.kirimin.mitsumine.db

import com.activeandroid.Model
import java.util.Calendar
import java.util.Locale

import com.activeandroid.query.Delete
import com.activeandroid.query.Select
import me.kirimin.mitsumine.model.Feed

public class FeedDAO private() {
    companion object {

        public fun save(feed: Feed) {
            val cal = Calendar.getInstance(Locale.JAPAN)
            feed.saveTime = cal.getTime().getTime()
            feed.save()
        }

        public fun findAll(): List<Feed> {
            return Select().from(javaClass<Feed>()).execute<Feed>()
        }

        public fun findByType(type: String): List<Feed> {
            return Select().from(javaClass<Feed>()).where("type = ?", type).execute<Feed>()
        }

        public fun deleteOldData() {
            val cal = Calendar.getInstance(Locale.JAPAN)
            val fiveDays = 1000 * 60 * 60 * 24 * 3.toLong()
            Delete().from(javaClass<Feed>()).where("saveTime < ? AND type = ?", cal.getTime().getTime() - fiveDays, Feed.TYPE_READ).execute<Model>()
        }
    }
}
